pinc/unicode.inc

```diff
@@ -170,3 +193,43 @@ function guess_string_encoding($text)
     # Give up and return ISO-8859-1
     return 'ISO-8859-1';
 }
+
+function make_title($codepoints)
+{
+    // make title from array of base codepoint and possible combining marks
+    $title = IntlChar::charName($codepoints[0]);
+    $count = count($codepoints);
+    if($count > 1)
+    {
+        $title .= " with " . IntlChar::charName($codepoints[1]);
+    }
+    for($index = 2; $index < $count; $index++)
+    {
+        $title .= " and " . IntlChar::charName($codepoints[$index]);
+    }
+    return $title;
+}
+
+function extended_to_hex($char)
+{
+    // find the components of extended character as hex strings
+    $codepoints = [];
+    preg_match_all("/./u", $char, $matches);
+    foreach($matches[0] as $match)
+    {
+        $codepoints[] = str_pad(dechex(IntlChar::ord($match)), 4, "0", STR_PAD_LEFT);
+    }
+    return implode(" ", $codepoints);
+}
+
+function extended_strlen($string)
+{
+    // how many 'extended Unicode sequences'?
+    return preg_match_all("/\X/u", $string);
+}
+
+function is_extended($string)
```

Comment:
So this returns a true or false if the string strictly contains only combined characters?

Reply:
Yes, there could be a better way of fixing the problem it was intended to solve. PQC checks for an initial '&' to distinguish the sort of problem. Althought it is totally unrealistic I put in & with a combining mark in the test project to see what would happen.